### DISCUSSION TITLE
Why is "DTSTART;VALUE=DATE" instead of DATE-TIME when IsAllDay = true?

### DISCUSSION THREAD

**Discussion by thehaseebahmed** [ical-org/ical.net#679] [Q&A] [answered]
This might be my lack of understanding of the iCal format since I'm new to the library and the format in general but why does setting `IsAllDay = true` on an Event result in the final iCal file having just the date instead of a DateTime? The reason why I ask this is because I'm in a UTC+1 timezone and when storing dates as UTC, the missing Time/TZ info from the final iCal files results in my events being shown 1 day earlier.

I've tried everything possible but the only hack that seems to work is setting "IsAllDay = false" and setting both the start/end dates to be the start/end of the given date. I don't like it and feels like a hack to me so there surely must be a better way.

Thanks!

**Comment by axunonb** [comment, marked as answer]
The `DTSTART` property in iCalendar specifies the start date and time for a calendar component, such as an event (`VEVENT`), a to-do (`VTODO`), or free/busy time (`VFREEBUSY`). By default, the value type for `DTSTART` is `DATE-TIME`, which includes both the date and the time.

When `VALUE=DATE` is specified, it means that the `DTSTART` property only includes the date, without any time component. This is used for all-day events or tasks that span an entire day. For example:

```
DTSTART;VALUE=DATE:20240101
```

This indicates that the event starts on January 1, 2024, and lasts the entire day¹(https://icalendar.org/iCalendar-RFC-5545/3-8-2-4-date-time-start.html)²(https://www.kanzaki.com/docs/ical/dtstart.html).

ical.net follows the [RFC 5545 specification](https://www.rfc-editor.org/rfc/rfc5545).
